packages/autocomplete-js/src/__tests__/detachedMediaQuery.test.ts

```diff
@@ -1,3 +1,71 @@
+import { autocomplete } from '../autocomplete';
+
 describe('detachedMediaQuery', () => {
-  test.todo('tests');
+  const originalMatchMedia = window.matchMedia;
+  const addEventListener = jest.fn();
+
+  beforeAll(() => {
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: true,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+        addEventListener,
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+  });
+
+  afterAll(() => {
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: originalMatchMedia,
+    });
+  });
+
+  test('fallback to the deprecated `addListener` if `addEventListener` is undefined', () => {
+    const addListener = jest.fn();
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: true,
+        media: query,
+        onchange: null,
+        addListener,
+        removeListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+    autocomplete<{ label: string }>({
+      id: 'autocomplete',
+      detachedMediaQuery: '',
+      container,
+      getSources() {
+        return [
+          {
+            sourceId: 'testSource',
+            getItems() {
+              return [{ label: 'Item 1' }];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
```

Comment:
We can use [`createSource`](https://github.com/algolia/autocomplete/blob/3e785660d65d568e611542dec8de20eb87a001b0/test/utils/createSource.ts) to remove some test boilerplate.

Reply:
Done in https://github.com/algolia/autocomplete/pull/600/commits/369e91a8b8ef5d229eec6788d4ebdc9a8f66cd26